cirq/study/resolver.py

```diff
@@ -90,7 +90,7 @@ def value_of(self,
             The value of the parameter as resolved by this resolver.
         """
         # Input is a float, no resolution needed: return early
-        if isinstance(value, float):
+        if isinstance(value, (float, np.float32)):
```

Comment:
Is there a reason not to have `int` here as well? In that case, could move this tuple of "pass-through" types to a module level variable and reuse it below.

Reply:
What about `np.int`? Also, why just 32-bit numpy floats? Perhaps this tuple should be a constant to ensure it's consistent throughout and easily updated.

Reply:
Alright, I went down a rabbit hole and made it work for numbers but also catered for sympy constants, and provided extensive tests for the cases that we care about. I think this is now a bit clearer and more robust than before.